### DISCUSSION TITLE
Failed to start electron app using pnpm

### DISCUSSION THREAD

**Discussion by holazz** [eclipse-theia/theia#11961] [General] [answered]
### Bug Description:

I'm using pnpm as package manager and an error occurs when starting electron app. I have checked that all extensions are using the same version and created a [minimal reproduction](https://github.com/holazz/theia-pnpm-bug), can anyone help me? Thanks in advance.

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/23100055/205865848-b98c4ddf-e387-43cd-aea2-abc1db8015af.png">

### Steps to Reproduce:

1. checkout this [repo](https://github.com/holazz/theia-pnpm-bug)
2. run `pnpm i && pnpm start:electron`

### Additional Information

- Operating System: macOS 11.4
- Node Version: 16.15.0
- Theia Version: 1.32.0
- PNPM Version: 7.18.0

`electron-app/package.json`:

```json
{
  "private": true,
  "name": "electron-app",
  "version": "0.0.0",
  "main": "src-gen/frontend/electron-main.js",
  "dependencies": {
    "@theia/core": "1.32.0",
    "@theia/electron": "1.32.0",
    "@theia/preferences": "1.32.0"
  },
  "devDependencies": {
    "@theia/cli": "1.32.0",
    "electron": "^15.5.7"
  },
  "scripts": {
    "prepare": "pnpm build",
    "build": "theia build --mode development",
    "start": "theia start"
  },
  "theia": {
    "target": "electron"
  }
}

```

**Comment by vince-fugnitto** [comment, marked as answer]
@holazz thank you for the discussion, we don't officially support `pnpm` as a package manager (we support [yarn](https://github.com/eclipse-theia/theia/blob/master/doc/Developing.md#prerequisites)) but if you want to troubleshoot the error occurs when multiple versions of theia-extensions are being used.

With `yarn` we could perform:

```
$ yarn theia check:theia-version
```

or

```
$ yarn why @theia/core
```

To identify what extension(s) is/are being pulled at different versions.

**Reply by holazz** [reply]
Gotcha! Thank you for the detailed explanation.
